R/build-logo.R

```diff
@@ -7,19 +7,36 @@ build_logo <- function(pkg = ".") {
 
   file_copy_to(pkg, logo_path, from_dir = path_dir(logo_path))
 
-  if (!requireNamespace("magick", quietly = TRUE)) {
-    message("magick not available, using default pkgdown favicon.ico")
+  cat_line("Creating ", dst_path("favicon.ico"))
 
-    if (!file_exists(path(pkg$dst_path, "favicon.ico"))) {
-      file_copy(path_pkgdown("assets", "favicon.ico"), pkg$dst_path)
-    }
-    return()
-  }
+  logo_string <- readBin(logo_path, what = "raw", n = 99999) %>%
+    openssl::base64_encode()
 
-  cat_line("Creating ", dst_path("favicon.ico"))
-  magick::image_read(logo_path) %>%
-	  magick::image_scale("32x32") %>%
-	  magick::image_write(path(pkg$dst_path, "favicon.ico"), format = "png")
+  template <- find_template("config", "favicon", ext = ".json")
+
+  data <- list(
+    url = pkg$meta$url,
+    logo_string = logo_string
+  )
+
+  json_favicon <- render_template(template, data)
```

Comment:
I think it would be cleaner to use generate the json from a list. Could you try doing that? I'm happy to get you started if it's not obvious what to do.